### PR TITLE
Capture HTTP 402 statuses

### PR DESCRIPTION
**Pull request by garyb**
For slamdata/slamdata#1000

### CODE REVIEW

File: src/Quasar/Error.purs [hunk 1]
```diff
@@ -24,22 +24,26 @@ import Control.Monad.Eff.Exception (Error, error, message)
...
   show (Error err) = "(Error " <> show err <> ")"
 
 printQError ∷ QError → String
-printQError NotFound = "Resource not found"
-printQError Forbidden = "Resource is unavailable, authorization is required first"
-printQError (Error err) = message err
+printQError = case _ of
+  NotFound → "Resource not found"
+  Forbidden → "Resource is unavailable, authorization is required first"
+  PaymentRequired → "Resource is unavailable, payment is required to use this feature"
```
Comment (garyb) [inline src/Quasar/Error.purs:43]: @jdegoes have you got any thoughts on how this message could be phrased better?

I think we'll be handling it manually in SlamData so can use a better message there anyway, but it's probably worth having something decent here too.

Comment (jdegoes) [inline src/Quasar/Error.purs:43]: This seems reasonable.

**Comment by garyb** [comment]
I've updated this to include 401 handling too, as the error for 403 was misleading, according to [what the statuses are used for](https://github.com/slamdata/quasar-advanced/tree/a5b0b8fcd8ee247c02770ebaf51a6c3e68681bf8#making-an-authenticated-request).

**Comment by garyb** [comment]
@beckyconning can you review this please?

**Pull request merged**